validation-test/Reflection/reflect_Optional_Bool.swift

```diff
@@ -0,0 +1,35 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-build-swift -g -lswiftSwiftReflectionTest %s -o %t/reflect_Optional_Bool
+// RUN: %target-codesign %t/reflect_Optional_Bool
+
+// RUN: %target-run %target-swift-reflection-test %t/reflect_Optional_Bool | %FileCheck %s --check-prefix=CHECK-%target-ptrsize
+
+
+// REQUIRES: reflection_test_support
+// REQUIRES: executable_test
+// UNSUPPORTED: use_os_stdlib
+
+import SwiftReflectionTest
+
+let value: Bool? = false
+reflect(enum: value)
```

Comment:
Since there are only three values -- true, false, nil -- I would copy this three times so we exercise every possibility.

We also want to check the _value_, not just the type.

Reply:
Agreed. While we're at it, do we also want to check `Bool??` just to be sure?

Reply:
@tbkka good call! There *is* a bug when reflecting optional bools after all. Both `true` and `false` reflect as nil:

```
Enum value:
(enum_value name=none index=1)
```

And nil fails:

```
Enum value:
swift_reflection_projectEnumValue failed.
```

This is most likely the bug I'm seeing in LLDB as well.

Reply:
Huzzah!  Once you have a test case that reproduces the bug, the rest should be easy!  😁